src/main/java/com/google/sps/server/ScheduleCalendar.java

```diff
@@ -62,30 +62,45 @@ public ScheduleCalendar(com.google.api.services.calendar.Calendar client) throws
    * Puts schedule onto secondary calendar
    * 
    * @param schedule the schedule to add to the calendar 
-   * @param calendarId the id of the secondary calendar to put the schedule on
    */
   public void addSchedule(Schedule schedule) throws IOException {
     try {
-      // get current date, so can add the courses to the appropriate days of the current month 
+      // get current calendar, so can add the courses to the appropriate days of the current month 
       java.util.Calendar calendar = java.util.Calendar.getInstance();
-      int nextSun = CalendarUtils.getDateOfNextSunday(calendar);
-      int month = CalendarUtils.getCurrentMonth(calendar);
-      int year = CalendarUtils.getCurrentYear(calendar);
 
-      // for each course in the schedule, add an event to the calendar
+      // for each course in the schedule, add all section times to calendar
       for (ScheduledCourse currentCourse : schedule.getCourses()) {
-        List<TimeRange> sectionTimes = currentCourse.getSection().getMeetingTimes();
-        for (int i = 0; i < sectionTimes.size(); i++) {
-          String startTime = CalendarUtils.calculateStartTime(sectionTimes.get(i), month, year, nextSun);
-          String endTime = CalendarUtils.calculateEndTime(sectionTimes.get(i), month, year, nextSun);
-          addEvent(currentCourse, startTime, endTime);
+        List<TimeRange> lectureSectionTimes = currentCourse.getLectureSection().getMeetingTimes();
+        addSectionToCalendar(lectureSectionTimes, calendar, currentCourse);
+        if (currentCourse.getLabSection() != null) {
+          List<TimeRange> labSectionTimes = currentCourse.getLabSection().getMeetingTimes();
+          addSectionToCalendar(labSectionTimes, calendar, currentCourse);
         }
       } 
     } catch (IOException e) {
       System.out.println("Couldn't add schedule");
     }   
   }
 
+   /**
+   * Helper function to turn all meeting times for a course section into calendar events
+   *
+   * @param sectionTimes
+   * @param calendar
+   * @param currentCourse 
+   */
+  private void addSectionToCalendar(List<TimeRange> sectionTimes, java.util.Calendar calendar, ScheduledCourse currentCourse) throws IOException {
+    try {
+      for (int i = 0; i < sectionTimes.size(); i++) {
+        String startTime = CalendarUtils.calculateStartTime(sectionTimes.get(i), calendar);
+        String endTime = CalendarUtils.calculateEndTime(sectionTimes.get(i), calendar);
+        addEvent(currentCourse, startTime, endTime);
+      }
+    } catch (IOException e) {
+      System.out.println("Couldn't iterate through section times");
```

Comment:
Please update this error message, it's a little misleading. The exception isn't caused by not being able to iterate through the section times, but rather that the section couldn't be added to the calendar.

Reply:
Thanks for pointing that out, that's true.